h/streamer/filter.py

```diff
@@ -29,11 +29,25 @@
 }
 
 
+def _normalize_uri_multi(items):
+    """
+    Normalise uri's for comparison.
+
+    :param items: A list or single item
+    :return: A list of mapped values, or a single value (if passed)
+    """
+    if isinstance(items, list):
+        return [normalize_uri(item) for item in items]
+
+    return normalize_uri(items)
+
+
 class FilterHandler:
-    FIELD_GETTERS = {
-        "/uri": attrgetter("target_uri"),
-        "/references": attrgetter("references"),
-        "/id": attrgetter("id"),
+    FIELDS = {
+        # Fields we accept mapped to a getter, and a normalization function
+        "/uri": [attrgetter("target_uri"), _normalize_uri_multi],
+        "/references": [attrgetter("references"), None],
+        "/id": [attrgetter("id"), None],
```

Comment:
You could use identity lambdas here, but it's about the same speed, and this is less weird